batch_get_item.go

```diff
@@ -29,6 +29,9 @@ func (e *BatchGetItemExpectation) WithError(err error) *BatchGetItemExpectation
 
 // BatchGetItem - this func will be invoked when test running matching expectation with actual input
 func (e *MockDynamoDB) BatchGetItem(input *dynamodb.BatchGetItemInput) (*dynamodb.BatchGetItemOutput, error) {
+	e.dynaMock.lock.Lock()
+	defer e.dynaMock.lock.Unlock()
+
```

Comment:
This is introducing behaviour that is very different from the underlying library that it's trying to mock.

We need to reconsider how we've structured out tests perhaps rather than make this kind of change.  Also we can look at ways to make this thread safe without using a pattern as heavy as a lock.

Reply:
@MS-TextNow found a lot better way to not have tests breaking each other.
There was actually a global Dynamock instance that was rewritten to everytime the `New` method was called.

Reply:
Nice! Yes that seems much safer.

Did you review why they might have been using a global var like that?  Curious if any of the existing methods required it in some abstract way.

Also are there Unit tests on this? I don't see any CI running on this repo.

Reply:
To be honest, I think this is at the origin a developer mistake.
Looked at it and it's used nowhere else in the code that the `New` function.
I ran the very limited test that there is on the repo present in the `.travis` folder. It's basically running the example folder and it passes. Also tested manually on our company repo and my tests do not have race condition anymore.

Reply:
Cool, I just reviewed it too.  I concur.

Great fix here!